### PR TITLE
Update to Ember 1.12.1, Ember CLI 0.2.7, add support for enabling ember-devtools in specific environments

### PR DESCRIPTION
**Pull request by aortbals**
This gives the application control over when this addon is included in the build. Most consumers will only want this in development, so that is the default.

This also changes the configuration structure to a more modern ember addon convention that puts a `ember-devtools` key at the root of `config/environment`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,16 +11,21 @@ A collection of useful functions for developing Ember apps. Best served from the
...
+        global: true,
+        enabled: environment === 'development'
+      }
+    }
     ```
-    You can then access the `devTools` functions globally (eg. you can run `routes()` in the console).
+    Setting `global` will allow access to the `devTools` functions globally (eg. you can run `routes()` in the console).
 
-    If you'd prefer these functions to be under a prefix set `emberDevTools: {global: 'foo'}` for `foo.routes()`.
+    The `enabled` option will enable the addon. This addon will only be included in the `development` environment by default.
```
Comment (aexmachina) [inline README.md]: "By default, this addon will only be included..."

File: config/environment.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
-module.exports = function(/* environment, appConfig */) {
-  return { };
+/**
+ * Export the default config for ember-devtools. By default, enable only
+ * in development.
+ */
+module.exports = function(environment, appConfig) {
+  appConfig['ember-devtools'] = appConfig['ember-devtools'] || {};
+  appConfig['ember-devtools'].enabled = appConfig['ember-devtools'].enabled || environment === 'development';
+  appConfig['ember-devtools'].global = appConfig['ember-devtools'].global || false;
```
Comment (aexmachina) [inline config/environment.js]: Better to write this as

```js
appConfig['ember-devtools'] = Ember.merge(appConfig['ember-devtools'] || {}, {
  enabled: environment === 'development',
  global: false
});
```

File: index.js [hunk 3]
```diff
@@ -2,5 +2,10 @@
 'use strict';
 
 module.exports = {
-  name: 'ember-devtools'
+  name: 'ember-devtools',
+
+  isEnabled: function isEnabled() {
+    var addonConfig = this.app.project.config(this.app.env)['ember-devtools'];
```
Comment (aexmachina) [inline index.js]: Is it possible for config to be a non-object here?

**Comment by aexmachina** [comment]
All good except for comments above and failing test.

**Comment by aexmachina** [comment]
My guess is that failure is because of PhantomJS. You may find that adding ember-cli-babelto `dependencies` in package.json (and removing ember-cli-6to5) fixes the problem.

File: index.js (hunk 3 above)
Comment (aortbals) [inline index.js]: I don't believe so, this is how it is done in the Ember CLI codebase, as well as several other addons I looked at.

File: config/environment.js (hunk 2 above)
Comment (aortbals) [inline config/environment.js]: I completely agree, however this is in a node environment and I don't believe we can use `Ember` here. The alternative would be to include something like [`lodash/object/merge`](https://github.com/ember-cli/ember-cli/blob/7af15c5f5e3ddbf8cdbde84d5b968a053c8c322d/lib/models/project.js#L14) or vendor https://github.com/emberjs/ember.js/blob/v1.12.0/packages/ember-metal/lib/merge.js.

Comment (aexmachina) [inline config/environment.js]: True, object-assign should do the trick.

**Comment by aortbals** [comment]
@aexmachina ended up having to do a little more on this than I initially planned to get the tests working. It is now on latest Ember and Ember CLI and the tests are up-to-date. This also fixes a few bugs that were caused by breaking API changes in Ember.

**Comment by aexmachina** [comment]
Awesome, nice one @aortbals. I'll have a look now.

**Comment by aexmachina** [comment]
Very nice, thanks!

**Comment by aexmachina** [comment]
Pushed v3.2.0

**Pull request merged**